include/simdjson/dom/document_stream.h

```diff
@@ -142,8 +184,8 @@ class document_stream {
   /** The error returned from the stage 1 thread. */
   error_code stage1_thread_error{UNINITIALIZED};
   /** The thread used to run stage 1 against the next batch in the background. */
-  std::thread stage1_thread{};
-
+  friend struct stage1_worker;
+  std::unique_ptr<stage1_worker> worker{new stage1_worker()};
```

Comment:
@jkeiser This is where I took care of your observation.

Reply:
Ooh! That works :) ... though I think for noexcept support, we need to:

1. use new (std::nothrow) (I don't know what this does but I see it elsewhere and it seems important)
2. Set document_stream.error = MEMALLOC if the new fails

This is definitely a reason to want to store this in the parser, though: it's essentially allocation. TBH, so is the thread itself.

Reply:
(My desire to have it in the parser does not factor into this patch, though, this is fine as it is. We already acquire a thread, what is a little more allocation between friends?)

Reply:
Done. I have added the nothrow and made it so that we will return a MEMALLOC should it fail. Note that it is kind of unlikely. You are more likely to fail when we create a thread.

Reply:
It is, yeah. Just a discipline thing to keep us truly noexcept safe.